CIPs/cip-120.md

```diff
@@ -0,0 +1,109 @@
+
+All pubsub messages for streams that include a value for the `sep` field in the genesis header MUST use the following topic:
+
+```markdown
+/ceramic/<network-name>/<separator-key>/<separator-value>
+```
+
+- `network-name` - the name of the Ceramic network used
+- `separator-key` - the value of the `sep` separator key field
+- `separator-value` - the value of the field specified by the `sep` field, encoded as `base36`
+
+### Example
+
+If a stream has the following genesis header,
+
+```javascript
+{
+  header: {
+    sep: 'model',
+    model: <bytes-k2t6wyfsu4pg1hnttc39koxlw0pdn1kemzrn33cdfhrbooozvo7wb14sa1uqwi>
+  },
+  ...
+}
+```
+
+it would use the following pubsub topic on main net:
+
+```markdown
+/ceramic/mainnet/model/k2t6wyfsu4pg1hnttc39koxlw0pdn1kemzrn33cdfhrbooozvo7wb14sa1uqwi
+```
+
+### Streams without `sep`
+
+For backwards compatibility if a stream doesn't have a `sep` field or no value for the field specified by `sep`,  will use the following pubsub topic:
+
+```markdown
+/ceramic/<network-name>/
+```
+
+Which is the pubsub topic currently used by Ceramic.
+
+### Peer discovery
+
+While gossipsub used by Ceramic enables ambient discovery of peers, an initial set of peers already in a topic is required to recieve messages. It is therefore recommended that peers announce to the libp2p DHT that they are subscribed to a given pubsub topic. Peers can thus also discover other peers by looking up the DHT records for the given topic. The key that should be used for DHT providing and lookup MUST be `sha2-256(separator-key || separator-value)`.
+
+
+## Rationale
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+By separating tip gossip into multiple pubsub topics, each node only needs to hear about a smaller subset of messages. Each node will subscribe only to the topics that are relevant for the streams it currently cares about.
+
+
+## Backwards Compatibility
+<!--All CIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The CIP must explain how the author proposes to deal with these incompatibilities. CIP submissions without a sufficient backwards compatibility section may be rejected outright.-->
+
+There are two approaches for how to transition to this new approach for pubsub topics.
+
+### Duplicate messages
+
+For a transition period new nodes will publish updates in both the old main topic as well as the new topic. This ensures backwards compatibility in the system at the cost of some amount of time with, at worst, the amount of messages.
```

Comment:
is the last sentence incomplete?  Is it meant to say at worst the amount of messages *doubled*?